<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PCBMillingGeneratorClass</class>
 <widget class="QMainWindow" name="PCBMillingGeneratorClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>913</width>
    <height>633</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PCB Milling Generator and Autolevel</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QRenderArea" name="renderArea" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>913</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionConvertGerberToGCode"/>
    <addaction name="actionOpen_GCODE_File"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>32</height>
    </size>
   </property>
   <property name="baseSize">
    <size>
     <width>0</width>
     <height>32</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen_GCODE_File"/>
   <addaction name="actionZoom_to_Fit"/>
   <addaction name="separator"/>
   <addaction name="actionExit"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="statusTip">
    <string/>
   </property>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>193</width>
     <height>560</height>
    </size>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QListWidget" name="lstFile">
       <property name="contextMenuPolicy">
        <enum>Qt::CustomContextMenu</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionConvertGerberToGCode">
   <property name="text">
    <string>Convert GERBER to GCode</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="../resources/MCB.qrc">
     <normaloff>:/MCBGenerator/Resources/exit.png</normaloff>:/MCBGenerator/Resources/exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionOpen_GCODE_File">
   <property name="icon">
    <iconset resource="../resources/MCB.qrc">
     <normaloff>:/MCBGenerator/Resources/fileopen.png</normaloff>:/MCBGenerator/Resources/fileopen.png</iconset>
   </property>
   <property name="text">
    <string>Open GCODE File</string>
   </property>
  </action>
  <action name="actionZoom_to_Fit">
   <property name="icon">
    <iconset resource="../resources/MCB.qrc">
     <normaloff>:/MCBGenerator/Resources/viewmagfit.png</normaloff>:/MCBGenerator/Resources/viewmagfit.png</iconset>
   </property>
   <property name="text">
    <string>Zoom to Fit</string>
   </property>
   <property name="toolTip">
    <string>Zoom to Fit</string>
   </property>
  </action>
  <action name="actionClose_File">
   <property name="icon">
    <iconset resource="../resources/MCB.qrc">
     <normaloff>:/MCBGenerator/Resources/fileclose.png</normaloff>:/MCBGenerator/Resources/fileclose.png</iconset>
   </property>
   <property name="text">
    <string>Close File</string>
   </property>
   <property name="toolTip">
    <string>Close Selected File</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QRenderArea</class>
   <extends>QWidget</extends>
   <header>qrenderarea.h</header>
   <container>1</container>
   <slots>
    <slot>clearRenderArea()</slot>
    <slot>ZoomToFit()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/MCB.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>PCBMillingGeneratorClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>456</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen_GCODE_File</sender>
   <signal>triggered()</signal>
   <receiver>PCBMillingGeneratorClass</receiver>
   <slot>OpenGcodeFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>456</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom_to_Fit</sender>
   <signal>triggered()</signal>
   <receiver>renderArea</receiver>
   <slot>ZoomToFit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>554</x>
     <y>323</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lstFile</sender>
   <signal>customContextMenuRequested(QPoint)</signal>
   <receiver>PCBMillingGeneratorClass</receiver>
   <slot>ShowContextMenuForListFile(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>208</y>
    </hint>
    <hint type="destinationlabel">
     <x>236</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lstFile</sender>
   <signal>itemChanged(QListWidgetItem*)</signal>
   <receiver>PCBMillingGeneratorClass</receiver>
   <slot>ListFileItemChanged(QListWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>257</y>
    </hint>
    <hint type="destinationlabel">
     <x>332</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClose_File</sender>
   <signal>triggered()</signal>
   <receiver>PCBMillingGeneratorClass</receiver>
   <slot>CloseSelectedFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>456</x>
     <y>315</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lstFile</sender>
   <signal>itemSelectionChanged()</signal>
   <receiver>PCBMillingGeneratorClass</receiver>
   <slot>ListFileItemSelectionChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>155</x>
     <y>228</y>
    </hint>
    <hint type="destinationlabel">
     <x>178</x>
     <y>-13</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionConvertGerberToGCode</sender>
   <signal>triggered()</signal>
   <receiver>PCBMillingGeneratorClass</receiver>
   <slot>ShowGerberToGCodeDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>456</x>
     <y>316</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OpenGcodeFile()</slot>
  <slot>ShowContextMenuForListFile(QPoint)</slot>
  <slot>ListFileItemChanged(QListWidgetItem*)</slot>
  <slot>CloseSelectedFile()</slot>
  <slot>ListFileItemSelectionChanged()</slot>
  <slot>ShowGerberToGCodeDialog()</slot>
 </slots>
</ui>
